### DISCUSSION TITLE
Is there a way to prevent trigger of the completion on newline/return character (when enter is pressed)?

### DISCUSSION THREAD

**Discussion by abzrgzdh** [saghen/blink.cmp#740] [Q&A] [answered]
Completion menu pops up when I press enter even in an empty line or when completion menu is not shown. I tried a bunch of combination of completion.trigger settings and only `show_on_keyword = true,` which totally defeats the purpose; I want to have completion menu automatically showing up except when enter is pressed (and no item is selected in the completion menu). Here's my config:

```lua
    opts = {
      appearance = {
        use_nvim_cmp_as_default = true,
        nerd_font_variant = 'mono'
      },

      snippets = {
        expand = function(snippet) require('luasnip').lsp_expand(snippet) end,
        active = function(filter)
          if filter and filter.direction then
            return require('luasnip').jumpable(filter.direction)
          end
          return require('luasnip').in_snippet()
        end,
        jump = function(direction) require('luasnip').jump(direction) end,
      },

      completion = {
        -- 'foo_|_bar' will match 'foo_' for 'prefix' and 'foo__bar' for 'full'
        keyword = { range = 'full', },

        -- Insert completion item on selection, don't select by default
        list = { selection = 'auto_insert' },

        -- Show documentation when selecting a completion item
        documentation = {
          auto_show = true,
          auto_show_delay_ms = 0,
          window = {
            border = "rounded",
          }
        },

        -- Display a preview of the selected item on the current line
        ghost_text = { enabled = false },

        menu = {
          border = "rounded",
          -- winhighlight = "Normal:Normal",
          draw = {
            columns = { { 'kind_icon' }, { 'label' } },
          }
        },

        trigger = {
          show_on_keyword = true,
          show_on_trigger_character = true,
          show_on_x_blocked_trigger_characters = { "'", '"', '(', '\n', '\r' },
        },
      },

      sources = {
        default = { 'lsp', 'path', 'luasnip', 'buffer' },

        -- Disable cmdline completions
        cmdline = {},
      },

      signature = { enabled = true },

      keymap = {
        preset = "enter",
        ["<Tab>"] = { "select_next", "snippet_forward", "fallback" },
        ["<S-Tab>"] = { "select_prev", "snippet_backward", "fallback" },
        -- ['<ESC>'] = { 'hide', 'fallback' },
      }
    },

```

**Comment by saghen** [comment]
What's your LSP?

**Reply by abzrgzdh** [reply]
Thank you for your reply. It’s not necessarily related to an LSP. It happens in all files, even in unnamed new buffers. The candidates include snippets, buffer completion candidates for all file types, and additionally LSP candidates (e.g., `clangd`).

**Comment by Hashino** [comment]
setting `sources.min_keyword_length` to not 0 solves it but then completions don't show on trigger character

**Comment by gbprod** [comment, marked as answer]
I had this issue using `blink.cmp` with [tpope/vim-eunuch](https://github.com/tpope/vim-eunuch/blob/master/plugin/eunuch.vim#L499), removing this plugin fixed for me.
Spotted that using `:imap <cr>` command.

**Comment by abzrgzdh** [comment]
Thank you @Hashino and @gbprod for your help. @gbprod, In my case it was the [tpope/vim-endwise](https://github.com/tpope/vim-endwise). It definitely does something with `<cr>` :).

**Comment by avonmoll** [comment]
I have this same question. When I run `:imap <cr>` I get `@<Plug>delimitMateCR`, but in my case I want to continue using DelimitMate.

**Comment by richin13** [comment]
Same, I have [tpope/vim-endwise](https://github.com/tpope/vim-endwise) and I'd like to keep using it. This is a recent bug introduced to blink.cmp because I had no issues a couple of weeks ago

**Reply by abzrgzdh** [reply]
There's another treesitter plugin that I'm using called [nvim-treesitter-endwise](https://github.com/RRethy/nvim-treesitter-endwise).

**Reply by richin13** [reply]
Nice, will try it out. Thanks!

**Comment by Hashino** [comment]
figured it out that it was `lexima.vim` for me. turning it off solved it. now I need to find a good replacement

**Reply by abzrgzdh** [reply]
I use [nvim-autopairs](https://github.com/windwp/nvim-autopairs)
